app/src/main/java/org/oppia/android/app/home/HomeViewModel.kt

```diff
@@ -107,20 +115,37 @@ class HomeViewModel(
   }
 
   /**
-   * Returns a [HomeItemViewModel] corresponding to the promoted stories to be displayed for this learner
-   * (see [PromotedStoryListViewModel]), or null if this profile does not have any promoted stories.
-   * Promoted stories are determined by any recent stories started by this profile.
+   * Returns a [HomeItemViewModel] corresponding to the promoted stories[PromotedStoryListViewModel] and Upcoming topics
+   * [ComingSoonTopicListViewModel]  to be displayed for this learner or null if this profile does not have any promoted stories.
+   * Promoted stories are determined by any recent stories or suggested stories started by this profile.
    */
-  private fun computePromotedStoryListViewModel(
-    ongoingStoryList: OngoingStoryList
+  private fun computeRecommendedActivityListViewModel(
+    recommendedActivityList: RecommendedActivityList
   ): HomeItemViewModel? {
-    val storyViewModelList = computePromotedStoryViewModelList(ongoingStoryList)
-    return if (storyViewModelList.isNotEmpty()) {
-      return PromotedStoryListViewModel(
-        activity,
-        storyViewModelList
-      )
-    } else null
+    when (recommendedActivityList.recommendationTypeCase) {
+      RecommendedActivityList.RecommendationTypeCase.RECOMMENDED_STORY_LIST -> {
+        val storyViewModelList =
+          computePromotedStoryViewModelList(recommendedActivityList.recommendedStoryList)
+        return if (storyViewModelList.isNotEmpty()) {
+          return PromotedStoryListViewModel(
+            activity,
+            storyViewModelList,
+            recommendedActivityList
+          )
+        } else null
+      }
+      RecommendedActivityList.RecommendationTypeCase.COMING_SOON_TOPIC_LIST -> {
+        val comingSoonTopicsList =
+          computeComingSoonTopicViewModelList(recommendedActivityList.comingSoonTopicList)
+        return if (comingSoonTopicsList.isNotEmpty()) {
+          return ComingSoonTopicListViewModel(
+            activity,
+            comingSoonTopicsList
+          )
+        } else null
+      }
+      else -> return null
```

Comment:
is this approach looks good? 
can we have something other than returning null or if we can modify the logic a bit here?

Reply:
I think that approach is right because if the list is null or empty we have to show nothing on the home screen. Promoted stories view will be hidden.